src/main/java/uk/nhs/hee/trainee/details/service/EventPublishService.java

```diff
@@ -55,6 +61,21 @@ public void publishProfileCreateEvent(TraineeProfile profile) {
     log.info("Sending profile creation event for trainee id '{}'", profile.getTraineeTisId());
 
     ProfileCreateEvent event = new ProfileCreateEvent(profile.getTraineeTisId());
-    messagingTemplate.convertAndSend(queueUrl, event);
+    messagingTemplate.convertAndSend(eventQueueUrl, event);
+  }
+
+  /**
+   * Publish a CoJ signed event.
+   *
+   * @param programmeMembership The signed {@link ProgrammeMembership}.
+   */
+  public void publishCojSignedEvent(ProgrammeMembership programmeMembership) {
+    log.info("Sending CoJ signed event for programme membership id '{}'",
+        programmeMembership.getTisId());
+
+    ConditionsOfJoining conditionsOfJoining = programmeMembership.getConditionsOfJoining();
```

Comment:
It is assumed that the CoJ will not be null, based on the code upstream - but we could add an additional check here if needed (and only send the message if its not null).

Reply:
I think this is a fair assumption, it might also be something we want to allow to happen in case we ever need to send a null to tell TIS to "unsign" a CoJ in the future.